.github/system_tests/test_daemon.py

```diff
@@ -0,0 +1,137 @@
+"""Tests to run with a running daemon."""
+import subprocess
+import sys
+import operator
+import time
+import numpy as np
+
+from aiida import orm
+from aiida.engine.daemon.client import get_daemon_client
+from aiida.orm import load_node
+from aiida.engine import launch
+from aiida.common import exceptions
+
+from aiida_optimize.engines import Bisection
+from aiida_optimize import OptimizationWorkChain
+import sample_processes
+
+TIMEOUTSECS = 4 * 60  # 4 minutes
+
+
+def print_daemon_log():
+    """Print daemon log."""
+    daemon_client = get_daemon_client()
+    daemon_log = daemon_client.daemon_log_file
+
+    print(f"Output of 'cat {daemon_log}':")
+    try:
+        print(subprocess.check_output(
+            ['cat', f'{daemon_log}'],
+            stderr=subprocess.STDOUT,
+        ))
+    except subprocess.CalledProcessError as exception:
+        print(f'Note: the command failed, message: {exception}')
+
+
+def jobs_have_finished(pks):
+    """Check if jobs with given pks have finished."""
+    finished_list = [load_node(pk).is_terminated for pk in pks]
+    node_list = [load_node(pk) for pk in pks]
+    num_finished = len([_ for _ in finished_list if _])
+
+    for node in node_list:
+        if not node.is_terminated:
+            print(f'not terminated: {node.pk} [{node.process_state}]')
+    print(f'{num_finished}/{len(finished_list)} finished')
+    return False not in finished_list
```

Comment:
`return all(finished_list)` is clearer IMO

Reply:
The method is not used as you mentioned below.